source/GameObject.cpp

```diff
@@ -21,13 +21,24 @@
 
 GOUID GameObject::idTracker = 0;
 
+bool GameObject::IsDestroy() const noexcept
```

Comment:
That function returns the state of the flag `destroy`. IsDestroyed() would imply `nullptr` since the objects with the `destroy` flag set will be removed from memory at the end of the frame/cycle/Destroyer call

Reply:
Got it that makes sense. It still smells off to me, but I don't have a better idea so maybe that's just a symptom of my own neurosis.